Linux support (FTV-143)

cpredmann opened the pull request:
This adds Linux support for the package, along with a CMake build configuration for the Swig library (only Linux supported currently).  The CMake is optionally able to build Mono and the USD.NET.dll, but this is disabled by default.  None of the Windows or OSX configurations have changed, and this was tested on 2018.3 and 2019.1.

jcowles commented:
Hrm, if this is based on dev, why did the Yamato config fix need to be pulled in?

cpredmann commented:
This had been based off of master, which is why it was missing that (along with 19.05 updates).  I've rebased against dev, and updated the Linux binary and CMake to use 19.05

clusty commented on cmake/CMakeLists.txt:
```diff
@@ -0,0 +1,286 @@
+set(USD_VERSION "v19.05" CACHE STRING "USD version (git tag)")
+
+if (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    set(boost_url "https://downloads.sourceforge.net/project/boost/boost/1.55.0/boost_1_55_0.tar.gz")
```
2 different boost versions for: Windows/Unix

cpredmann replied:
That's just following along with the versions from the USD build script (https://github.com/PixarAnimationStudios/USD/blob/38c70fb869c01228f82fd844a8c420860b5131c1/build_scripts/build_usd.py#L519), but it would be nice to figure out why they're not moving everything to VFX CY2018 (at least).

clusty commented:
Is the current Howto up to date on how to update the USD version ?

jcowles commented:
I don't know CMake well, but I'm assuming Vlad does since he commented on it.

And yeah, if you wouldn't mind updating the README.md and BUILDING.MD, I think that's all that's left.

Thanks Chris!!

cpredmann commented:
Sure thing!  I made a couple of edits to the README and BUILDING md files, if you want to take a look over...

jcowles commented:
(I just rebased this branch to dev, FYI)
